pycqed/analysis_v2/readout_analysis.py

```diff
@@ -1340,14 +1340,18 @@ class MultiQubit_SingleShot_Analysis(ba.BaseDataAnalysis):
             readouts for a single readout then n_readouts has to include them.
         channel_map: dictionary with qubit names as keys and channel channel
             names as values.
-        thresholds: dictionary with qubit names as keys and threshold values as
-            values.
+
     Optional options in the options_dict:
         observables: Dictionary with observable names as a key and observable
             as a value. Observable is a dictionary with name of the qubit as
-            key and boolean value indicating if it is selecting exited states.
+            key and boolean value indicating if it is selecting excited states.
             If the qubit is missing from the list of states it is averaged out.
         readout_names: used as y-axis labels for the default figure
+        thresholds: dictionary with qubit names as keys and threshold values as
+            values. If not specified, 'shots_thresholded' must be passed in.
+        shots_thresholded (dict): single shots thresholded keyed by qubit name.
+        shot_filter: 1D boolean array: filters the shots before averaging
+            (eg. to remove f-level detected states)
```

Comment:
why not implement it the same way as is done in MultiQubit_TimeDomain_Analysis, as a lambda function data_filter

Reply:
Both work, I guess. Lambda functions cannot be stored easily in hdf5 files, but are more consise. Is it a critical point? If yes, I can adapt it.